test/lit/passes/optimize-instructions-cmps.wast

```diff
@@ -0,0 +1,168 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: wasm-opt %s --optimize-instructions -S -o - | filecheck %s
+
+;; Tests for "x < 0 || x > POSITIVE_CONST   ==>   unsigned(x) > POSITIVE_CONST" optimization
+
+(module
+ ;; CHECK:      (global $g (mut i32) (i32.const 0))
+ (global $g (mut i32) (i32.const 0))
+
+ ;; CHECK:      (func $cmp (param $0 i32) (result i32)
+ ;; CHECK-NEXT:  (i32.gt_u
+ ;; CHECK-NEXT:   (local.get $0)
+ ;; CHECK-NEXT:   (i32.const 255)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $cmp (param $0 i32) (result i32)
+  (i32.or
```

Comment:
Please add comments in the tests, like here, "we can optimize this to a single i32.gt_u".